lib/open_food_network/orders_and_fulfillments_report/customer_totals_report.rb

```diff
@@ -210,6 +212,27 @@ def payment_fee(order)
         order.adjustments.select{ |a| a.originator_type == 'Spree::PaymentMethod' }.
           map(&:amount).sum.to_f
       end
+
+      def amount_with_adjustments(line_items)
+        line_items.map do |line_item|
+          price_with_adjustments(line_item) * line_item.quantity
+        end.sum
+      end
+
+      def price_with_adjustments(line_item)
+        return 0 if line_item.quantity == 0
+        (line_item.price + line_item_adjustments(line_item).map(&:amount).sum / line_item.quantity).
+          round(2)
+      end
+
+      def line_item_adjustments(line_item)
+        line_item.order.adjustments.select{ |a| a.source_id == line_item.id }
+      end
+
+      def shipping_method(line_items)
+        line_items.first.order.shipments.first.
+          andand.shipping_rates.andand.first.andand.shipping_method
+      end
```

Comment:
same as for orders but worse, I dont think this is acceptable, it's adding variations to a system (adjustments) that is already a mess...

we need at least to move all this to line_item_decorator, we need to consider merging/replacing existing methods in line_item_decorador.

Also, we should probably consider creating new services for these methods (and the existing ones in the order and line_item models):
- in order : order_adjustments_service
- here: line_item_adjustments_service

Reply:
This method was needed because now we are eager-loading more associated data, and need to access: `line_items.first.order.shipments.first.shipping_rates.first.shipping_method` in multiple places, but in the result set any of these links in the chain of associations can be `nil`.